From 0080d6f109d769478b049a0b3621183f8df71776 Mon Sep 17 00:00:00 2001
From: HolonProduction <holonproduction@gmail.com>
Date: Sat, 31 Jan 2026 13:40:51 +0100
Subject: [PATCH] LSP: Handle clients that do not support `CompletionContext`
References: https://github.com/godotengine/godot/issues/115631
            https://github.com/godotengine/godot/pull/115672
Upstream: merged

less errors shown for communication between kate and godot via lsp

can be probably dropped if included in next release

---
 modules/gdscript/language_server/godot_lsp.h | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/modules/gdscript/language_server/godot_lsp.h b/modules/gdscript/language_server/godot_lsp.h
index 31a5bdc14320..a249a7a5d620 100644
--- a/modules/gdscript/language_server/godot_lsp.h
+++ b/modules/gdscript/language_server/godot_lsp.h
@@ -1438,7 +1438,7 @@ struct CompletionContext {
 	/**
 	 * How the completion was triggered.
 	 */
-	int triggerKind = CompletionTriggerKind::TriggerCharacter;
+	int triggerKind = CompletionTriggerKind::Invoked;
 
 	/**
 	 * The trigger character (a single character) that has trigger code complete.
@@ -1461,7 +1461,10 @@ struct CompletionParams : public TextDocumentPositionParams {
 
 	void load(const Dictionary &p_params) {
 		TextDocumentPositionParams::load(p_params);
-		context.load(p_params["context"]);
+
+		if (p_params.has("context")) {
+			context.load(p_params["context"]);
+		}
 	}
 
 	Dictionary to_json() {
